script.js

```diff
@@ -1,4 +1,632 @@
+// Stage 1: Cards are shuffled. Player starts game and is given 2 cards and dealer is given 2 cards
+// Stage 2: Player decides if they want to 'Hit' a card or to 'Stand', dealer automatically draws if it does not have 21 points. If the player 'bust' the game ends, and the results is tallied. If the player 'Hit' and has not bust, stage 2 is repeated.
+// Stage 3: If the player decides to stand, the results is tallied. Game is resetted.
+
+var gameMode = "gameStart";
+var playerHand = [];
+var dealerHand = [];
+var playerScore = 0;
+var dealerScore = 0;
+var currentIndex = 0;
+var playerTotalWins = 0;
+var dealerTotalWins = 0;
+var myOutputValue = "Error in states";
+let timerInterval; // for alert popup
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (input == "restart") {
+    gameMode = "gameStart";
+    playerTotalWins = 0;
+    dealerTotalWins = 0;
+  }
+
+  if (gameMode == "gameStart") {
+    playerHand = [];
+    dealerHand = [];
+
+    // deep clone of array
+    var copiedDeck = [...newCardDeck];
```

Comment:
This is a shallow clone actually, because changing a card inside of `copiedDeck` will also change the card in `newCardDeck`. This is because JavaScript objects also work like arrays, in terms of the differences between values and references (memory). However, changing the ORDER of `copiedDeck` should not change the order of `newCardDeck`, and vice versa.

Think of this in 2 levels (top-most to bottom-most): (1) deck (array), (2) card (object). A shallow copy means any changes to (1) in the original should not affect the (1) of the shallow copied versions. In a deep copy, changes to (1) and (2) in the original should not affect the (1) and (2) of the shallow copied versions.

I guess you do not really need deep copies for Blackjack, unless you want the card objects to be different in `copiedDeck` and `newCardDeck` for any reason. If you are really interested, we have some [Bootcamp notes on how you can do deep copies](https://bootcamp.rocketacademy.co/0-0-language-and-tooling/0.2-es6/0.2.4-reference-vs.-value#reference-vs.-value). 

Despite the hack mentioned in the aforementioned article, there isn't really a proper way to do deep copies efficiently. If you happen to have special values in objects or arrays such as `null`, or `Infinity`, they will mess up the deep copy hack. 

I see that you are comfortable enough to install `npm` packages yourself, so if you ever want to deep clone arrays/objects with special values, you can try using `lodash` with its `cloneDeep` functionality. You can read about `cloneDeep` [here](https://www.geeksforgeeks.org/lodash-_-clonedeep-method/). `lodash` is just an open-source library with a lot of helper functions for a lot of common programming tasks that do not come pre-packaged with regular JavaScript.